fix: Add documentation (#1567)
embeddings-benchmark/mteb: docs/api/task.md

# Tasks

> A task is an implementation of a dataset for evaluation. It could for instance be the MIRACL dataset consisting of queries, a corpus of documents 
> as well as the correct documents to retrieve for a given query. In addition to the dataset a task includes specification for how a model should be run on the dataset and how its output should be evaluation. Each task also come with extensive metadata including the license, who annotated the data and so on.

Reviewer: ```suggestion
A task is an implementation of a dataset for evaluation. It could, for instance, be the MIRACL dataset consisting of queries, a corpus of documents 
,and the correct documents to retrieve for a given query. In addition to the dataset, a task includes the specifications for how a model should be run on the dataset and how its output should be evaluated. Each task also comes with extensive metadata including the license, who annotated the data, etc.
```

<figure markdown="span">
    ![](../images/visualizations/task_explainer.png){ width="80%" }
    <figcaption>An overview of the tasks within `mteb`</figcaption>
</figure>

## Utilities

:::mteb.get_tasks

:::mteb.get_task

## Metadata

> Each task also contains extensive metadata, we annotate this using the following object. This allows to use [pydantic](https://docs.pydantic.dev/latest/) to evaluate that the 
> metadata is valid and consistent. 

Reviewer: ```suggestion
Each task also contains extensive metadata. We annotate this using the following object, which allows us to use [pydantic](https://docs.pydantic.dev/latest/) to validate the metadata. 
```
